### PR TITLE
Set ALLOW_MEMORY_GROWTH=1

### PR DESCRIPTION
**Pull request by mdboom**
An alternative to #154.  This would be preferred if it works and doesn't have a significant performance penalty.

### CODE REVIEW

**Comment by mdboom** [comment]
So tests are passing.  That's a good sign.

I'd like to add a test based on the issue that @teonbrooks was having when running out of memory before merging this just for confirmation that it works as advertised.

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,8 @@ LDFLAGS=\
 	-s MODULARIZE=1 \
 	$(CPYTHONROOT)/installs/python-$(PYVERSION)/lib/libpython$(PYMINOR).a \
   -s "BINARYEN_METHOD='native-wasm'" \
-  -s TOTAL_MEMORY=536870912 \
+  -s TOTAL_MEMORY=1073741824 \
```
Comment (mdboom) [inline Makefile]: In order to get the test to path, you need more memory to start with as well.

File: test/conftest.py [hunk 2]
```diff
@@ -214,16 +214,29 @@ def run_web_server(q):
...
+        def is_cgi(self):
+            if self.path.startswith('/test/') and self.path.endswith('.cgi'):
+                self.cgi_info = '/test', self.path[6:]
+                return True
+            return False
+
+        def log_message(self, *args, **kwargs):
+            pass
+
+    Handler.extensions_map['.wasm'] = 'application/wasm'
```
Comment (mdboom) [inline test/conftest.py]: This adds CGI support so we can generate data on-the-fly.

File: test/test_pandas.py [hunk 3]
```diff
@@ -16,3 +16,22 @@ def test_extra_import(selenium, request):
...
+
+    selenium.load_package("pandas")
+    selenium.load_package("matplotlib")
+
+    selenium.run("""
+        import pyodide
+        import matplotlib.pyplot as plt
+        import pandas as pd
+
+        pd.read_json(pyodide.open_url('test/largish.json.cgi'))
```
[the remark this answers is not in the document]
Comment (mdboom) [reply test/test_pandas.py]: It isn't taking the full extra 500MB -- it's just that the memory size works best with powers of 2.

The real issue, IMHO, is that just importing the code alone is taking up nearly 500MB.  That's why I think approaches like #152 may really help.  Not only does that keep the files compressed in memory, it's stores them in an ArrayBuffer outside of the Wasm heap, so it doesn't count against this value (expandable or not).

**Comment by teonbrooks** [comment]
thanks @mdboom. is the plugin link updated with these changes?

**Comment by mdboom** [comment]
Looks to be, but I haven't tried.  (It's deploying from master automatically now...)

**Pull request merged**